tensorflow/compiler/tf2xla/kernels/image_resize_ops.cc

```diff
@@ -816,4 +816,43 @@ void ResizeBilinearGradOp::Compile(XlaOpKernelContext* ctx) {
 
 REGISTER_XLA_OP(Name("ResizeBilinearGrad"), ResizeBilinearGradOp);
 
+ResizeNearestNeighborGradOp::ResizeNearestNeighborGradOp(
+    OpKernelConstruction* ctx)
+    : XlaOpKernel(ctx) {
+  OP_REQUIRES_OK(ctx, ctx->GetAttr("align_corners", &align_corners_));
+  OP_REQUIRES_OK(ctx, ctx->GetAttr("half_pixel_centers", &half_pixel_centers_));
+  OP_REQUIRES(ctx, !half_pixel_centers_ || !align_corners_,
+              errors::Unimplemented("If half_pixel_centers is True, "
+                                    "align_corners must be False."));
+}
+
+void ResizeNearestNeighborGradOp::Compile(XlaOpKernelContext* ctx) {
```

Comment:
`ResizeBilinearGradOp` falls back to the TF implementation on GPU do we need to do the same here?

https://github.com/tensorflow/tensorflow/blob/c208c490902a4eab77f6d3f41463798d1ccc1051/tensorflow/compiler/tf2xla/kernels/image_resize_ops.cc#L763-L769

Reply:
Probably a good idea to try and do that.